pandas/core/generic.py

```diff
@@ -3739,7 +3739,6 @@ def _slice(self: FrameOrSeries, slobj: slice, axis=0) -> FrameOrSeries:
 
         Slicing with this method is *always* positional.
         """
-        assert isinstance(slobj, slice), type(slobj)
```

Comment:
this is not the patch; something is going down the wrong path and ending up here. this routine is *only* for slices.

Reply:
Ok, I will look into that. I just wanted to note that in version 1.0.9 it went down just the same path except that the exception wasn't there yet. In this sense we are not talking about a regression but about fixing a bug which seems to have been there since a while ago.

Reply:
@CloseChoice can you check if the suggestion from here https://github.com/pandas-dev/pandas/issues/35509#issuecomment-687430169 passes the tests you added?